internal/csi-addons/rbd/replication.go

```diff
@@ -937,3 +945,38 @@ func checkVolumeResyncStatus(ctx context.Context, localStatus types.SiteStatus)
 
 	return nil
 }
+
+func getCurrentReplicationStatus(ctx context.Context,
+	mirrorGlobalStatus types.GlobalStatus,
+) (replication.GetVolumeReplicationInfoResponse_Status, string, error) {
+	// get the state of replication
+	localStatus, err := mirrorGlobalStatus.GetLocalSiteStatus()
+	if err != nil {
+		log.ErrorLog(ctx, err.Error())
+
+		return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "", err
+	}
+
+	localState := localStatus.GetState()
+	desc := localStatus.GetDescription()
+
+	// check if mirroring state is up/down
+	isUp := localStatus.IsUP()
+
+	if !isUp {
+		return replication.GetVolumeReplicationInfoResponse_DEGRADED, desc, nil
+	}
+
+	switch localState {
+	case librbd.MirrorImageStatusStateUnknown.String():
+		return replication.GetVolumeReplicationInfoResponse_UNKNOWN, desc, nil
+	case librbd.MirrorImageStatusStateError.String():
+		return replication.GetVolumeReplicationInfoResponse_ERROR, desc, nil
+	case librbd.MirrorImageStatusStateReplaying.String():
+		return replication.GetVolumeReplicationInfoResponse_HEALTHY, desc, nil
+	case librbd.MirrorImageStatusStateStopped.String():
+		return replication.GetVolumeReplicationInfoResponse_HEALTHY, desc, nil
+	default:
+		return replication.GetVolumeReplicationInfoResponse_UNKNOWN, desc, nil
```

Comment:
instead of multiple return, set a variable and return at the end?

also add a unit testing for this one?